src/stores/MapOptionsStore.ts

```diff
@@ -96,6 +98,7 @@ const tfCycle: RasterStyle = {
     attribution:
         osmAttribution +
         ', <a href="https://www.thunderforest.com/maps/opencyclemap/" target="_blank">Thunderforest Cycle</a>',
+    tilePixelRatio: 2,
```

Comment:
This way the raster tiles aren't blurry like they were for Mapbox

Reply:
Isn't the avoidance of fractional numbers for zoom what the blurry avoids? `constrainResolution: true` See your comment https://github.com/openlayers/openlayers/issues/12963#issuecomment-961834559

And tilePixelRatio (additionally) improves blurry for high dpi devices? (but for mapbox we had these "`2x`" in the URL)

(then also the question would be: could we set it to 1 for low dpi devices somehow?)

Reply:
> Isn't the avoidance of fractional numbers for zoom that avoids the blurry look? (constrainResolution: true)
See your comment openlayers/openlayers#12963 (comment)

Yes, avoiding fractional zoom levels is part of it. But `tilePixelRatio` is still needed I think. As far as I know we need to tell OpenLayers also about the resolution of our tiles. From the documentation: https://openlayers.org/en/latest/apidoc/module-ol_source_XYZ-XYZ.html

> The pixel ratio used by the tile service. For example, if the tile service advertizes 256px by 256px tiles but actually sends 512px by 512px images (for retina/hidpi devices) then tilePixelRatio should be set to 2.

So we need to set this number depending on the tile service. 

> but for mapbox we had these "2x" in the URL

Yes, for some tile providers adding `2x` changes the resolution of the tiles they send, but either way we need to tell OpenLayers the resolution of the tiles we feed to it?

> (then also the question would be: could we set it to 1 for low dpi devices somehow?)

Do you see blurry tiles on some device, or what do you want to improve here?

Reply:
Also see this and the following comments: https://github.com/graphhopper/graphhopper-maps/issues/159#issuecomment-973917832